File: Models/Wsi/PickTicket.cs
namespace intranet.Models.Wsi
{
    public class PickTicket
    {
        public string PickTicketNumber { get; set; }
        
        public string OrderNumber { get; set; }

        public string Action { get; set; }

        public int Store { get; set; }

        public Address Customer { get; set; }

        public Address Recipient { get; set; }

        public string ShippingMethod { get; set; }

        public List<LineItem> LineItems { get; set; }

        public DateTime OrderDate { get; set; }

        public int Channel { get; set; }

        public DateTime CreatedAt { get; set; }

        public DateTime UpdatedAt { get; set; }

        public class Address
        {
            public string Name { get; set; }

            public string Street { get; set; }

            public string City { get; set; }

            public string State { get; set; }

            public string Country { get; set; }

            public string Zip { get; set; }
        }

        public class LineItem
        {
            public string PickTicketNumber { get; set; }

            public int LineNumber { get; set; }

            public string Action { get; set; }

            public string Sku { get; set; }

            public int Units { get; set; }

            public int UnitsToShip { get; set; }

            public DateTime CreatedAt { get; set; }

            public DateTime UpdatedAt { get; set; }
        }
    }
}

File: Client/src/views/Wsi/Order/PickTicket.js
import PropTypes from 'prop-types';
import { React } from 'react';

import Address from 'components/Address';
import Product from 'views/Wsi/Order/Product';

function PickTicket(props) {
  const pickTicket = props.pickTicket;
  const orderDate = new Date(pickTicket.orderDate);
  const orderCreationDate = new Date(pickTicket.createdAt + 'Z');
  return (
    <div>
      <div>
        <p>Order Number: {pickTicket.orderNumber}</p>
        <p>Order Date: {orderDate.toLocaleDateString()}</p>
        <p>WSI Order Creation Date: {orderCreationDate.toLocaleString()}</p>
        <p>Shipping Method: {pickTicket.shippingMethod}</p>
      </div>

      <div className='horizontal-flex address-container'>
        <div>
          <h2>Customer</h2>
          <Address address={pickTicket.customer} />
        </div>
        <div>
          <h2>Recipient</h2>
          <Address address={pickTicket.recipient} />
        </div>
      </div>
      <h2>Products</h2>
      {pickTicket.lineItems.map(lineItem => {
        return <Product product={lineItem} key={lineItem.sku}/>;
      })}
    </div>
  );
}

PickTicket.propTypes = {
  pickTicket: PropTypes.object
};

export default PickTicket;
